### ISSUE TITLE
NeoVim buffer switching slow when using vim-airline

### ISSUE THREAD

**Issue by geomaster** [neovim/neovim#4524]
- Neovim version: NVIM v0.1.3-380-gc8d830e
- Operating system: Arch Linux with Plasma 5 desktop (Linux 4.4.5-1-ARCH #1 SMP PREEMPT x86_64)
- Terminal emulator: Konsole 15.12.3

### Actual behaviour

When vim-airline is loaded, switching buffers (`:bnext`, `:bprevious`) incurs a delay which disappears when vim-airline is not present. I believe this is a NeoVim issue as the same problem doesn't exist with Vim. The delay is similar when running `nvim` from Konsole, as well as when using `nvim-qt`. With Vim, the switching is instantaneous when running `vim` from Konsole, as well as when running `gvim`.

### Expected behaviour

As with Vim, vim-airline should not be able to significantly slow down buffer switching.

### Steps to reproduce

1. Map a key (e.g. Tab) to a buffer switching command: `:nmap <Tab> :bnext<CR>`
2. Install vim-airline and leave it with the default options.
3. Open two buffers.
4. Hold down the Tab key, forcing NeoVim to rapidly switch between the two buffers.
5. After holding the key for 30 seconds or so, release it and observe how long the switching persists after the key is released.

Now repeat these steps but without enabling vim-airline. For me, with it enabled, NeoVim keeps switching the buffers for ~5 seconds after releasing the Tab key, while with vim-airline disabled, the switching stops instantaneously. Vim behaves fine in this case regardless of whether vim-airline is present.

### Remarks

If you believe this to be a vim-airline issue, I'll be happy to file it with that project. It seems to me that this issue is the same as #2219, however, the problem there went away when NeoVim was recompiled in the Release build type. However, my NeoVim is already built like this:

```
NVIM v0.1.3-380-gc8d830e
Build type: Release
Compilation: /usr/share/perl5/vendor_perl/auto/share/dist/Cope/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -Wconversion -O2 -DNDEBUG -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -D_GNU_SOURCE -I/tmp/yaourt-tmp-geomaster/aur-neovim-git/src/neovim-git/build/config -I/tmp/yaourt-tmp-geomaster/aur-neovim-git/src/neovim-git/src -I/usr/include -I/usr/include -I/usr/include/luajit-2.0 -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/tmp/yaourt-tmp-geomaster/aur-neovim-git/src/neovim-git/build/src/nvim/auto -I/tmp/yaourt-tmp-geomaster/aur-neovim-git/src/neovim-git/build/include
Compiled by geomaster@desktop.geo

Optional features included (+) or not (-): +acl   +iconv    +jemalloc 
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

I'm not aware of any methods to precisely measure the buffer switching time (I'm not acquainted with the codebase enough to know if there are any profiling tools or to insert mine), so let me know if there are any and I'll be happy to report back the numbers.

**Comment by wsdjeg** [comment]
yes ,I get same issue with the latest version of neovim.I can confirm it.

**Comment by wdv4758h** [comment]
@geomaster @wsdjeg Does the lastest version of neovim still have this issue on your machine❓
I've try it on my machine with the lastest version of neovim, but I can't reproduce it. 😕

```
NVIM v0.1.5-280-g176f223
Build type: Release
Compilation: /usr/bin/clang -Wconversion -O2 -DNDEBUG -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/tmp/neovim-git/src/neovim-git/build/config -I/tmp/neovim-git/src/neovim-git/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/tmp/neovim-git/src/neovim-git/build/src/nvim/auto -I/tmp/neovim-git/src/neovim-git/build/include
Compiled by dv@linux-dv

Optional features included (+) or not (-): +acl   +iconv    +jemalloc 
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

vim-airline version https://github.com/vim-airline/vim-airline/commit/4d39cb6f2078326dd07f2ab680a0365299a2589e

**Comment by wsdjeg** [comment]
I still get th issue with airline,but can not repoduce by `nvim -u NONE`,hold on tab with the mapping `nmap <Tab> :bnext<CR>`, when stop hold on tab, the buffer still swich between two buffers, and it will stop after about 3-4 seconds.

**Comment by geomaster** [comment]
Turns out this was mostly me being a dork, or more precisely, still a bit of a Vim noob. I fixed the problem by doing `set hid` in NeoVim's vimrc. Turns out I had this in the vimrc that Vim used (not even realized it because I was using one of those pre-made vimrcs), which made the performance way better there. I guess that vim-airline gets some info about a file when it's first opened, and without `set hid`, essentially, every switching to a file buffer meant reopening it. As far as my problem is concerned, it's gone, so I'm going to go ahead and close the issue. A big thank you to everyone who chimed in with suggestions and help.
